### PR TITLE
Downstream 26539 & 28074 - proper handling of invalid config

### PR DESCRIPTION
**Pull request by MorrisJobke**
This is a downstream of https://github.com/owncloud/core/pull/28074 ~~and https://github.com/owncloud/core/pull/26539 to make it easy to test. owncloud/core#26539 is also needed for #6146  (to review: look into both commits separately)~~

### CODE REVIEW

File: apps/files_external/lib/Lib/Storage/InvalidStorage.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Files_External\Lib\Storage;
+
+use OCP\Files\StorageNotAvailableException;
+
+class InvalidStorage {
```
Comment (icewind1991) [inline apps/files_external/lib/Lib/Storage/InvalidStorage.php]: There is an `OC\Files\Storage\FailedStorage` that should probably bne reused instead

**Review by icewind1991 — CHANGES_REQUESTED** [review]

File: apps/files_external/lib/Lib/Storage/InvalidStorage.php (hunk 1 above)
Comment (MorrisJobke) [inline apps/files_external/lib/Lib/Storage/InvalidStorage.php]: I rebased this. Sadly I wasn't apply to make this change here happen. @icewind1991 Could you please help me here? Thanks

**Comment by MorrisJobke** [comment]
I removed the commits regarding the handling of storages and made a separate PR out of it. This is now ready for review.

**Comment by MorrisJobke** [comment]
Please re-review.

**Review by LukasReschke — APPROVED** [review]

**Review by rullzer — APPROVED** [review]

**Pull request merged**